packages/node/src/generators/application/lib/create-files.ts

```diff
@@ -68,4 +69,50 @@ export function addAppFiles(tree: Tree, options: NormalizedSchema) {
   if (options.js) {
     toJS(tree);
   }
+
+  // Update webpack config to include devtoolModuleFilenameTemplate for better debugging
+  if (
+    options.bundler === 'webpack' &&
+    options.addPlugin !== false &&
+    hasWebpackPlugin(tree)
+  ) {
+    const webpackConfigPath = joinPathFragments(
+      options.appProjectRoot,
+      'webpack.config.js'
+    );
+    if (tree.exists(webpackConfigPath)) {
```

Comment:
Instead of the string replacement logic here, can we update the template at `packages/node/src/generators/application/files/common/webpack.config.js__tmpl__`? If debug requires sourcemaps, maybe it should just be `sourceMaps: true` regardless of `NODE_ENV`.